### PR TITLE
Enforces the order of RoundRobinLoadBalancer when positive overflow

### PR DESCRIPTION
**Pull request by jizhuozhi**
#1074

### CODE REVIEW

**Comment by OlgaMaciaszek** [comment]
Fixes gh-1074.

File: spring-cloud-loadbalancer/src/main/java/org/springframework/cloud/loadbalancer/core/RoundRobinLoadBalancer.java [hunk 1]
```diff
@@ -36,6 +36,7 @@
  *
  * @author Spencer Gibb
  * @author Olga Maciaszek-Sharma
+ * @author jizhuozhi
```
Comment (OlgaMaciaszek) [inline spring-cloud-loadbalancer/src/main/java/org/springframework/cloud/loadbalancer/core/RoundRobinLoadBalancer.java:39]: Please change to full name and surname.

File: spring-cloud-loadbalancer/src/test/java/org/springframework/cloud/loadbalancer/core/RoundRobinLoadBalancerTest.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+import org.springframework.cloud.loadbalancer.support.SimpleObjectProvider;
+
+import static java.lang.Integer.MAX_VALUE;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+/**
+ * @author jizhuozhi
```
Comment (OlgaMaciaszek) [inline spring-cloud-loadbalancer/src/test/java/org/springframework/cloud/loadbalancer/core/RoundRobinLoadBalancerTest.java:35]: Please change to full name and surname.

**Review by OlgaMaciaszek — CHANGES_REQUESTED** [review]
Hello, @jizhuozhi thanks for moving the PR to `3.1.x`. As requested, please use your full name and surname in the author tag instead of GH username.

File: spring-cloud-loadbalancer/src/main/java/org/springframework/cloud/loadbalancer/core/RoundRobinLoadBalancer.java (hunk 1 above)
Comment (jizhuozhi) [reply spring-cloud-loadbalancer/src/main/java/org/springframework/cloud/loadbalancer/core/RoundRobinLoadBalancer.java:39]: OK

**Review by OlgaMaciaszek — APPROVED** [review]

**Pull request merged**